runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -2718,22 +2756,24 @@ MM_CopyForwardScheme::getNextScanCache(MM_EnvironmentVLHGC *env, UDATA preferred
 		}
 
 		if (NULL == cache) {
-			omrthread_monitor_enter(_scanCacheMonitor);
-			_scanCacheWaitCount += 1;
+			omrthread_monitor_enter(*_workQueueMonitorPtr);
+			*_workQueueWaitCountPtr += 1;
```

Comment:
It would be cleaner if getNextScanCache() and similar scanCache specific methods use directly scanCache monitor and waitCount directly. I guess, they would be used only in non-hybrid mode. 
But I hope we will get rid of those scanCache specific methods, eventually.

Reply:
I removed method getNextScanCache() and getNextScanCacheOnNode().